cmd/opampsupervisor/supervisor/supervisor.go

```diff
@@ -826,10 +815,18 @@ func (s *Supervisor) runAgentProcess() {
 			restartTimer.Reset(5 * time.Second)
 
 		case <-restartTimer.C:
+			s.logger.Debug("Agent starting after start backoff")
 			s.startAgent()
 
 		case <-s.healthCheckTicker.C:
 			s.healthCheck()
+
+		case <-s.doneChan:
+			err := s.commander.Stop(context.Background())
+			if err != nil {
+				s.logger.Error("Could not stop agent process", zap.Error(err))
+			}
+			return
```

Comment:
This change is due to a race condition I was bumping into with the e2e tests. Basically, the shutdown would happen while the command was in the middle of starting, so this goroutine would never exit (the command never exited), then Supervisor.Shutdown would never exit as a consequence. 

Instead, we just handle the full lifecycle of the command all in the same goroutine.